src/caf/space/zone_translation.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+Module containing ZoneTranslation class.
+
+Class for producing zone translations from a set of inputs, provided by the
+ZoneTranslationInputs class in 'inputs'.
+"""
+import logging
+import warnings
+import pandas as pd
+from caf.space import weighted_funcs as nf, zone_correspondence as zc, inputs as si
+
+##### CONSTANTS #####
+LOG = logging.getLogger(__name__)
+logging.captureWarnings(True)
+
+
+##### CLASSES #####
+class ZoneTranslation:
+    """
+    Store paramaters and create zone translations.
+
+    This is the main class for the caf.space tool. Running it with an
+    instance of 'inputs.ZoningTranslationInputs' will return a zone
+    translation dataframe.
+
+    Parameters
+    ----------
+    params: ZoningTranslationInputs
+        Params should usually be read in from a yml file using the
+        load_yaml method of the ZoningTranslationInputs class. Refer to
+        this class for info on parameters.
+
+    Returns
+    -------
+    Instance of self
+        Instance of ZoneTranslation, where a spatial or weighted
+        zone translation, dependent on inputs, is stored as
+        self.zone_translation
+    """
+
+    def __init__(self, params: si.ZoningTranslationInputs):
+        self.params = params
+        self.zone_1 = params.zone_1
+        self.zone_2 = params.zone_2
+        self.output_path = params.output_path
+        self.cache_path = params.cache_path
+        if params.lower_zoning:
+            self.lower_zoning = params.lower_zoning
+        if params.method:
+            self.method = params.method
+        self.tolerance = params.tolerance
+        self.rounding = params.rounding
+        self.filter_slithers = params.filter_slithers
+        self.run_date = params.run_date
+        sorted_names = sorted([params.zone_1.name, params.zone_2.name])
+        self.names = (sorted_names[0], sorted_names[1])
+        self.cacher = self.cache_path / f"{self.names[0]}_{self.names[1]}"
+        self.cacher.mkdir(exist_ok=True, parents=True)
```

Comment:
cache_path is the base path read in from the config. cacher is just another name for the subfolder written there for files to actually be saved in

Reply:
I'm maybe being a little fussy about naming here, but I'm a big believer that the name should be more descriptive rather than just something to assign a variable to.

I find `cacher` to be bit a bit of an ambiguous name, and it also implies that it's an class that does something rather than just being a path. i.e., that I could call `self.cacher.cache(item)` and it would cache my `item` somehow